rfcs/rfc-furnishings-job.md

```diff
@@ -0,0 +1,176 @@
+- Days of the week that a given stage can be run will be defined in `configuration` table
+
+
+### Stage 1
+
+1. Retrieve cron string from `DISSOLUTIONS_STAGE_1_SCHEDULE` in configuration table. This cron string stores which days of the week new dissolutions can be initiated.
+2. If the current day of the week is not a valid day(use `DISSOLUTIONS_STAGE_1_SCHEDULE` to determine this), skip remaining steps in stage 1.
+3. Find existing batches that are still in stage 1 of dissolution.
+4. For matching batches that do not have any furnishing entries yet, do the following for each batch:
+    1. Find all businesses where `business_processing.batch_id` = current batch id
+    2. For all business matches that have emails:
+        1. Create email furnishing entry
+        2. Queue a message to emailer.  For more details, refer to email section of this document.
+    3. For all business matches that don't have emails:
+        1. Create letter furnishing entry
+        2. Create and add letter to either AR or transition pdf.
+    4. Send AR and transition pdf to BCMail+.  For more details, refer to letter section of this document.
+5. For matching batches that have existing furnishing entries still in stage 1, do the following for each batch:
+    1. Find all businesses for current batch that can be sent a letter
+    2. For all business matches in batch, do the following:
+        1. Create letter furnishing entry for each business match
+        2. Create and add letter to either AR or transition pdf.
+    3. Send AR and transition pdfs to BCMail+.  For more details, refer to letter section of this document.
+
+
+### Stage 2
+
+1. Retrieve cron string from `DISSOLUTIONS_STAGE_2_SCHEDULE` in configuration table. This cron string stores which days of the week new dissolutions can be initiated.
+2. If the current day of the week is not a valid day(use `DISSOLUTIONS_STAGE_2_SCHEDULE` to determine this), skip remaining steps in stage 2.
+3. Find existing batches that are in stage 2 of dissolution.
+4. For matching batches that do not have any stage 2(`furnishings.name` = "INTENT_TO_DISSOLVE") furnishing entries yet, do the following for each batch:
+    1. Create intent to dissolve furnishing entries
+    2. Create required xml to send to gazette.
+    3. Send xml to gazette.  For more details, refer to gazette section of this document.
+
+
+### Stage 3
+
+1. Retrieve cron string from `DISSOLUTIONS_STAGE_3_SCHEDULE` in configuration table. This cron string stores which days of the week new dissolutions can be initiated.
+2. If the current day of the week is not a valid day(use `DISSOLUTIONS_STAGE_3_SCHEDULE` to determine this), skip remaining steps in stage 2.
+3. Find existing batches that are in stage 3 of dissolution.
+4. For matching batches that do not have any stage 3(`furnishings.name` = "CORP_DISSOLVED") furnishing entries yet, do the following for each batch:
+    1. Create intent to dissolve furnishing entries
```

Comment:
This looks like a duplicate of Stage 2.4.1. What's different here in Stage 3?

Reply:
Mostly the same.  The furnishing name and stage being filtered on are just different.

Reply:
Actually, you are right Sev.  

Didn't look closely enough.  It should be "Create business dissolved furnishing entries". 

Will get that updated!